### PR TITLE
webhook refactor

### PR DESCRIPTION
**Pull request by R-Lawton**
# Verification
. `make local-setup` with `MCTC_WORKLOAD_CLUSTERS_COUNT` set to bring up workload cluster(s)
. `make deploy`

What should be created:
* tls certs in the tls directory in webhook-config/control
* Webhook Service, Ingress and TLS secret in the control cluster
* webhook configs should be created in the workload clusters with the matching CA bundle as the new created certs (when decoded should be the tls.crt duplicated)

### CODE REVIEW

File: hack/local-setup.sh [hunk 1]
```diff
@@ -143,6 +145,21 @@ deployDashboard() {
...
+  clusterName=${1}
+  echo "Deploying the webhook configuration to (${clusterName})"
+
+  kubectl config use-context kind-${clusterName}
+
+  plain=$( cat $TLS_CERT_PATH/tls.crt $TLS_CERT_PATH/tls.crt  )
+  echo $WEBHOOK_PATH/webhook-configs.yaml
+  endocded=$( echo "$plain" | base64 )
+
+  yq -e  -i ".webhooks[0].clientConfig.caBundle =\"$endocded\"" $WEBHOOK_PATH/webhook-configs.yaml
```
Comment (sergioifg94) [inline hack/local-setup.sh:158]:
```suggestion
  encoded=$( echo "$plain" | base64 )

  yq -e  -i ".webhooks[0].clientConfig.caBundle =\"$encoded\"" $WEBHOOK_PATH/webhook-configs.yaml
```
Tiny typo

File: hack/local-setup.sh [hunk 2]
```diff
@@ -171,12 +188,17 @@ deployExternalDNS ${KIND_CLUSTER_CONTROL_PLANE}
 #6. Deploy argo cd
 deployArgoCD ${KIND_CLUSTER_CONTROL_PLANE}
 
-#7. Deploy Dashboard
+#7. Deploy webhook conifgs
+deployWebhookConfigs ${KIND_CLUSTER_CONTROL_PLANE}
```
Comment (sergioifg94) [inline hack/local-setup.sh:192]: I might be wrong but my understanding is that we will only need the webhook configs in the workload clusters

Comment (philbrookes) [reply hack/local-setup.sh:192]: Yeah, this makes sense to me. Webhook in the workload clusters, endpoint in the control plane. (Though we might want webhooks for gateways on the control plane in the future)

File: hack/local-setup.sh (hunk 1 above)
Comment (R-Lawton) [reply hack/local-setup.sh:158]: thank you nice catch

File: hack/local-setup.sh (hunk 2 above)
Comment (R-Lawton) [reply hack/local-setup.sh:192]: ok cool that makes sense, i misunderstood the flow of the script ill change that

**Review by philbrookes — APPROVED** [review]

**Pull request merged**